src/redux/sagas/expenditures/createExpenditure.ts

```diff
@@ -46,12 +44,23 @@ function* createExpenditure({
   const txChannel = yield call(getTxChannel, metaId);
   const batchKey = 'createExpenditure';
 
+  // Group slots by token address, this is useful as we need to call setExpenditurePayouts method separately for each token
+  type MapValue = ExpenditureSlotFieldValue & { id: number };
+  const slotsByTokenAddress = new Map<string, MapValue[]>();
+  slots.forEach((slot, index) => {
+    const currentSlots = slotsByTokenAddress.get(slot.tokenAddress) ?? [];
+    // Add id to each slot
+    const slotWithId = { ...slot, id: index + 1 };
+    slotsByTokenAddress.set(slot.tokenAddress, [...currentSlots, slotWithId]);
+  });
```

Comment:
This might look unusual at first glance, but we want to make one `setExpenditurePayouts` call per token address, so we're grouping the slots beforehand